diff --git a/MODULE.bazel b/MODULE.bazel
index 7e5e4566..4a26e691 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -95,7 +95,7 @@ git_override(
 #bazel_dep(name = "protobuf", version = "21.7", repo_name = "com_google_protobuf")
 bazel_dep(name = "grpc", version = "1.56.3", repo_name = "com_github_grpc_grpc")
 
-go_sdk = use_extension("@io_bazel_rules_go//go:extensions.bzl", "go_sdk")
+go_sdk = use_extension("@io_bazel_rules_go//go:extensions.bzl", "go_sdk", dev_dependency=True)
 go_sdk.download(
     name = "go_sdk",
     version = "1.21.8",
